### PR TITLE
[web][docs] Improve HtmlElementView widget docs.

### LINKED ISSUE #49634: [web] use HtmlElement inside a Widget like HtmlElementView in Flutter Web
The official Flutter Web example not longer working on Flutter Channel Dev.

**Link:** https://github.com/flutter/flutter_web/blob/master/examples/html_platform_view/lib/main.dart

**Example:**
```
import 'dart:html';

import 'package:flutter_web_ui/ui.dart' as ui;

import 'package:flutter_web/material.dart';

void main() {
  ui.platformViewRegistry.registerViewFactory(
      'hello-world-html',
      (int viewId) => IFrameElement()
        ..width = '640'
        ..height = '360'
        ..src = 'https://www.youtube.com/embed/IyFZznAk69U'
        ..style.border = 'none');

  runApp(Directionality(
    textDirection: TextDirection.ltr,
    child: SizedBox(
      width: 640,
      height: 360,
      child: HtmlElementView(viewType: 'hello-world-html'),
    ),
  ));
}
```

Current PlatformViewsRegistry only has the method **"getNextPlatformViewId"** and not longer **"registerViewFactory"**

**PlatformViewsRegistry - platform_views.dart:**
```
/// The [PlatformViewsRegistry] responsible for generating unique identifiers for platform views.
final PlatformViewsRegistry platformViewsRegistry = PlatformViewsRegistry._instance();

/// A registry responsible for generating unique identifier for platform views.
///
/// A Flutter application has a single [PlatformViewsRegistry] which can be accesses
/// through the [platformViewsRegistry] getter.
class PlatformViewsRegistry {
  PlatformViewsRegistry._instance();

  // Always non-negative. The id value -1 is used in the accessibility bridge
  // to indicate the absence of a platform view.
  int _nextPlatformViewId = 0;

  /// Allocates a unique identifier for a platform view.
  ///
  /// A platform view identifier can refer to a platform view that was never created,
  /// a platform view that was disposed, or a platform view that is alive.
  ///
  /// Typically a platform view identifier is passed to a [PlatformView] widget
  /// which creates the platform view and manages its lifecycle.
  int getNextPlatformViewId() => _nextPlatformViewId++;
}
```

**Flutter Doctor:**
```
[✓] Flutter (Channel dev, v1.14.5, on Mac OS X 10.14.5 18F132, locale en-US)
    • Flutter version 1.14.5 at /Users/username/Documents/flutter
    • Framework revision 92cbaa3efe (5 days ago), 2020-01-24 05:43:01 +0100
    • Engine revision 439a21816e
    • Dart version 2.8.0 (build 2.8.0-dev.5.0 e84bea25df)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/username/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.6.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Community Edition (version 2019.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 36.1.4
    • Dart plugin version 191.7830

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 79.0.3945.130
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

**Notes:**

There no current example/method to add **HtmlElement** into Widgets like **HtmlElementView** used to do thank to **"registerViewFactory"**:

```HtmlElementView(viewType:"string-key");```

### LINKED ISSUE #143922: Web regression: `HtmlElementView onPlatformViewCreated` fires before HTML element created
Prior to 3.19, when using `HtmlElementView`'s  `onPlatformViewCreated` callback, you could access the HTML element within the callback:

```dart
HtmlElementView(
    viewType: 'myView',
    onPlatformViewCreated: (viewId) async {
        // Should be able to access the created view here
        window.document.getElementById('myViewId');
    }
)
```

This is failing in Flutter 3.19, as the HTML element has not yet been added to the DOM at the time `onPlatformViewCreated` is called.

This has broken production code for me, and I believe it will have broken two projects I've previously worked on.

## Steps to reproduce

Run this code sample on Flutter web with 3.19.

**Expected results**: shows a red box (screenshot from 3.16.9): 
Screenshot 2024-02-22 at 10 06 57

**Actual results**: throws `Error: TypeError: null: type 'Null' is not a subtype of type 'DivElement'` because the div has not yet been added to the DOM.

```dart
import 'package:flutter/material.dart';
import 'dart:ui_web' as ui_web;
import 'dart:html' as html;

void main() {
  ui_web.platformViewRegistry.registerViewFactory(
    'myView',
    (int viewId, {Object? params}) {
      return html.DivElement()
        ..id = 'myDivId'
        ..style.width = '100px'
        ..style.height = '100px';
    },
  );

  runApp(
    MaterialApp(
      home: Scaffold(
        body: HtmlElementView(
          viewType: 'myView',
          onPlatformViewCreated: (viewId) async {
            final div = html.window.document.getElementById('myDivId')
                as html.DivElement;

            div.style.backgroundColor = 'red';
          },
        ),
      ),
    ),
  );
}
```

<details><summary>Logs</summary>

```console

Launching lib/main.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:53778/20k_NPLe51o=/ws
Debug service listening on ws://127.0.0.1:53778/20k_NPLe51o=/ws
Connecting to VM Service at ws://127.0.0.1:53778/20k_NPLe51o=/ws
Error: TypeError: null: type 'Null' is not a subtype of type 'DivElement'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3  throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                         _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                         _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1258:30                        _installSpecializedAsCheck
packages/web_platform_view/main.dart 23:17                                   <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54           runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5           _async
packages/web_platform_view/main.dart 21:35                                   <fn>
packages/flutter/src/widgets/_html_element_view_web.dart 65:30               <fn>
dart-sdk/lib/async/zone.dart 1661:54                                         runUnary
dart-sdk/lib/async/future_impl.dart 162:18                                   handleValue
dart-sdk/lib/async/future_impl.dart 838:44                                   handleValueCallback
dart-sdk/lib/async/future_impl.dart 867:13                                   _propagateToListeners
dart-sdk/lib/async/future_impl.dart 643:5                                    [_completeWithValue]
dart-sdk/lib/async/future_impl.dart 713:7                                    callback
dart-sdk/lib/async/schedule_microtask.dart 40:11                             _microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5                              _startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:7           <fn>
```

</details>

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.19.1, on macOS 14.3.1 23D60 darwin-arm64, locale en-GB)
    • Flutter version 3.19.1 on channel stable at /Users/tom/.puro/envs/stable/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision abb292a07e (33 hours ago), 2024-02-20 14:35:05 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/tom/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/tom/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.86.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.80.0

[✓] Connected device (4 available)
    • Tom’s iTelephone (mobile)  • 00008130-000A4D8820A1401C            • ios            • iOS 17.3.1 21D61
    • iPhone 15 Pro Max (mobile) • 2904497E-B3E2-450A-B364-A5A0B0D0901A • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-17-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 14.3.1 23D60 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 121.0.6167.184

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by ditman**
This PR expands the `HtmlElementView` widget DartDocs, with the following sections:

* Usage: How to use the widget, two ways:
  * The `HtmlElementView.fromTagName` constructor
  * The `PlatformViewRegistry` way
* Lifecycle: There's an `onCreated` callback on the widget. When does it get called?
* HTML Lifecycle: How to listen to events coming from the DOM.
* Visibility: what is the `isVisible` property for?

Small additional tweaks here and there to mention common pitfalls of using HtmlElementView on the web, and mentions to workarounds, like `package:pointer_interceptor`.

## Issues

* Fixes: https://github.com/flutter/flutter/issues/143922
* Fixes: https://github.com/flutter/flutter/issues/49634
* Related: https://github.com/flutter/website/issues/5063

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by ditman** [comment]
(Unsure what's going to happen with the `dart` samples, they might break because of their reliance in external libraries like package:web or dart:ui_web.)

**Review by mdebbar — APPROVED** [review]
LGTM ❤️

**Comment by ditman** [comment]
```
packages/flutter/lib/src/widgets/platform_view.dart:402:25: Undefined name 'onElementCreated' (statement) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:425:5: Undefined name 'ui_web' (statement) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:428:13: Undefined class 'HTMLDivElement' (statement) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:428:40: Undefined name 'web' (statement) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:454:30: Undefined name 'onElementCreated' (statement) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:455:23: Use 'const' literals as arguments to constructors of '@immutable' classes (statement) (prefer_const_literals_to_create_immutables)
packages/flutter/lib/src/widgets/platform_view.dart:511:28: Undefined class 'HTMLDivElement' (top-level declaration) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:512:13: Undefined class 'Element' (top-level declaration) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:512:36: Undefined name 'web' (top-level declaration) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:524:18: The name 'HTMLDivElement' isn't a type, so it can't be used in an 'as' expression (top-level declaration) (cast_to_non_type)
packages/flutter/lib/src/widgets/platform_view.dart:529:13: Undefined class 'ResizeObserver' (top-level declaration) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:529:43: Undefined name 'web' (top-level declaration) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:530:9: Undefined class 'JSArray' (top-level declaration) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:530:17: The name 'ResizeObserverEntry' isn't a type, so it can't be used as a type argument (top-level declaration) (non_type_as_type_argument)
packages/flutter/lib/src/widgets/platform_view.dart:531:9: Undefined class 'ResizeObserver' (top-level declaration) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:537:11: The function 'onElementInjected' isn't defined (top-level declaration) (undefined_function)
packages/flutter/lib/src/widgets/platform_view.dart:539:9: The getter 'toJS' isn't defined for the type 'Null Function(InvalidType, InvalidType)' (top-level declaration) (undefined_getter)
packages/flutter/lib/src/widgets/platform_view.dart:563:39: Undefined name 'context' (top-level declaration) (undefined_identifier)
packages/flutter/lib/src/widgets/platform_view.dart:565:11: Undefined class 'JSAny' (top-level declaration) (undefined_class)
packages/flutter/lib/src/widgets/platform_view.dart:565:32: Undefined name 'ui_web' (top-level declaration) (undefined_identifier)
Found 20 snippet code errors.
```

UGGHGGHGHGHHG

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 1]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// Both callbacks are called **after the HTML `element` has been created, but
+/// _before_ it's injected into the DOM.**
+///
+/// ### HTML Lifecycle
+///
+/// The Browser DOM APIs have additional HTML lifecycle callbacks for the root
+/// `element` of an `HtmlElementView`.
+///
+/// #### Element Injected Into The DOM
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:508]: nit here and elsewhere: I haven't seen "inject into" used in this context. Maybe "attach to"?

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 2]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// It is common for JS code to locate the DOM elements they need with a
+/// selector, rather than accepting said DOM elements directly. **In those cases,
+/// the `element` must be injected into the DOM for the selector to work.**
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:512]: nit: too much bold

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 3]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// The example below demonstrates **how to create an `onElementInjected` function**
+/// that gets called when the root `element` is injected into the DOM
+/// using a `ResizeObserver` through `package:web` from the `onElementCreated`
+/// lifecycle method:
+///
+/// ```dart
+/// import 'dart:js_interop';
+/// import 'package:web/web.dart' as web;
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:521]: This pay fail the analyzer. Code examples are supposed to compile, but `package:flutter` does not depend on `package:web`. We had this issue recently with another code example.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 4]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///   observer.observe(element);
+/// }
+/// ```
+///
+/// * Read more about [`ResizeObserver` in the MDN](https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API).
+///
+/// #### Other Observers
+///
+/// The example above uses a `ResizeObserver` because it can be applied directly
+/// to the `element` that is about to be injected. The most _correct_ observer
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:559]: If it's the most "correct", why are we putting it in the "Other" section? :)

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 5]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// import 'package:flutter/widgets.dart';
+///
+/// void useHostElement(BuildContext context) {
+///   final int flutterViewId = View.of(context).viewId;
+///   final JSAny? hostElement = ui_web.views.getHostElement(flutterViewId);
+///   // Use `package:web` with `hostElement`...
+/// }
+/// ```
+///
+/// **Important:** If your flutter App uses `registerViewFactory` to register a
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:582]: nit: "Flutter app" instead of "flutter App"

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 6]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// factory function that creates your `HtmlElementView` widget, take into
+/// account that that the `viewId` parameter passed to your factory function
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:584]: s/that that/that/

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 7]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// **is not the same** as the `flutterViewId` obtained above:
+///
+/// * `flutterViewId` (from `View.of(context)`) represents the [FlutterView]
+///   where the web app is currently rendering.
+/// * `viewId` (passed to the `viewFactory` function) represents a unique ID
+///   for the `HtmlElementView` instance that is being injected into the app.
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:590]: I'd shorten this paragraph to just one sentence: "Note that `FlutterView.viewId` and the platform `viewId` passed to `RegisterViewFactory` identify different kinds of objects".

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 8]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// Read more about [FlutterView] on Flutter's API docs:
+///
+/// * [`View.of`](https://api.flutter.dev/flutter/widgets/View/of.html)
+/// * [`getHostElement`](https://main-api.flutter.dev/flutter/dart-ui_web/FlutterViewManagerProxy/getHostElement.html)
+///
+/// ## Pointer events
+///
+/// `HtmlElementView` will interfere with Flutter's normal gesture detection. In
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:599]: I would not use the word "interfere". This is all part of the normal functionality. This effect is expected and desired.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 9]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// Flutter.
+///
+/// [`package:pointer_interceptor`](https://pub.dev/packages/pointer_interceptor)
+/// may help in some cases where Flutter content needs to be overlaid on top of
+/// an `HtmlElementView`. Alternatively, the `pointer-events: none` property can
+/// be set `onElementCreated`; but that will prevent **ALL** interactions with
+/// the underlying HTML content.
+///
+/// If the `HtmlElementView` is an `<iframe>` element, Flutter will not be able
+/// to see any pointer events that land in the `<iframe>` (click/tap, drag,
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:612]: nit: replace "not be able to see any" with "receive". BTW, is this always true? Can't you create an iframe with origin trust policies that allow observing cross-frame events?

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 10]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// drop...). In those cases, the HTML View will seem like it's "swallowing"
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:613]: nit: say ", etc" instead of "..."

Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:613]: nit: what is "HTML View"? Consider using one term to refer to platform views throughout the docs. Multiple terms referring to the same thing tend to be confusing.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 11]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// ## Visibility
+///
+/// `HtmlElementView` objects have an `isVisible` property that can be passed when
+/// registering the factory function, or calling `fromTagName`. In this case,
+/// the "visibility" refers to whether the `HtmlElementView` will paint pixels
+/// or not.
+///
+/// Correctly defining this value helps the Flutter web rendering engine optimize
+/// the amount of "overlays" it'll need to render a particular scene; some browsers
+/// can only have a limited number of overlays on a single page at a given time.
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:628]: It's safe to say "all" browsers. But more concretely, we should probably just say that overlays have performance cost both in memory and execution speed, and therefore it is best to reduce the number of overlays. `isVisible` is one way to do it.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 12]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// In general, `isVisible` should be left to its default value of `true`, but
+/// in some `HtmlElementView`s (like the `pointer_interceptor` or `Link` widget),
+/// it can be set to `false`, so the engine doesn't "waste" an overlay to render
+/// Flutter content on top of views that don't paint any pixels.
+///
+/// ## Accessibility
+///
+/// Flutter web enables accessibility by waiting for a double-tap in a
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:637]: nit: "Flutter Web` (for consistency)

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 13]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// button that covers the full page.
+///
+/// Placing this button in front of the scene would cause `HtmlElementView`s to
+/// not receive any pointer events, so the button is sent to the back.
+/// `HtmlElementView`s interferes with normal pointer events (See **Pointer
+/// events** above), so in order to enable accessibility, a **user must be able
+/// to double tap the app outside of the root `element` of an `HtmlElementView`**.
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:644]: I'd like to understand more about this. I wonder if we're documenting a bug here :)

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 14]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
 /// Make sure that your HTML views are sized no larger than necessary, or you
 /// may cause difficulty for users trying to enable accessibility.
 ///
-/// {@macro flutter.widgets.AndroidView.lifetime}
+/// Alternatively, a Flutter web app may programmatically enable accessibility
+/// by calling [SemanticsBinding.ensureSemantics].
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:650]: Not sure if this is the right place to document this 🤔

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 15]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// ----
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:652]: What do the empty line and the `----` do?

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 16]
```diff
@@ -419,6 +670,7 @@ class HtmlElementView extends StatelessWidget {
   ///
   /// [onElementCreated] is called when the DOM element is created. It can be
   /// used by the app to customize the element by adding attributes and styles.
+  /// This method is called *before* the element is injected into the DOM.
```
Comment (yjbanov) [inline packages/flutter/lib/src/widgets/platform_view.dart:673]: Not sure if this was meant to be a standalone paragraph like in the method below, but if it was, then there needs to be one empty line between paragraphs.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 17]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
   State<AppKitView> createState() => _AppKitViewState();
 }
 
-/// Callback signature for when the platform view's DOM element was created.
+/// Function that gets called when the platform view's DOM element is created.
 ///
 /// [element] is the DOM element that was created.
```
[the remark this answers is not in the document]
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:365]: Ah yes, this is a good point. I'll try to mention this in other places that we receive a generic Object.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 18]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// The example creates a `<div>` element, then calls the `onElementCreated`
+/// callback with the created `<div>`, so it can be customized **before it is
+/// injected into the DOM.**
+///
+/// (See more details about `onElementCreated` in the **Lifecycle** section below.)
+///
+/// ### Using the `PlatformViewRegistry`
+///
+/// The "classic" way to create HTML Elements in Flutter web is still available.
```
[the remark this answers is not in the document]
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:416]: I rephrased this as:

"The primitives used to implement `fromTagName` are available for general use through `dart:ui_web`..."

Instead of calling this "classic". It is "classic" in the sense that if anybody was doing platform views before this documentation came out, it's 99% possible that they were using this way.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 19]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+/// (See more details about `onPlatformViewCreated` in the **Lifecycle** section
+/// below.)
+///
+/// ## Lifecycle
+///
+/// `HtmlElementView` widgets behave like any other Flutter stateless widget, but
+/// with an additional lifecycle method: `onPlatformViewCreated` / `onElementCreated`
+/// (depending on the constructor, see **Usage** above).
+///
+/// Both callbacks are similar, but each one receives a different parameter:
```
[the remark this answers is not in the document]
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:492]: Rephrased as "The only difference between the two callbacks..." which is the bit of information I think matters, especially because I point out the similarity right below.

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 20]
```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
...
+///
+/// * `onPlatformViewCreated` will be called with the created `viewId` as a parameter,
+///   and needs `ui_web.platformViewRegistry.getViewById` to retrieve the created
+///   element (See [PlatformViewCreatedCallback]).
+/// * `onElementCreated` will be called with the created `element` directly,
+///   skipping its `viewId` (See [ElementCreatedCallback]).
+///
+/// Both callbacks are called **after the HTML `element` has been created, but
+/// _before_ it's injected into the DOM.**
```
[the remark this answers is not in the document]
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:501]: I've bolded `after` and `before`.

I'm using bold in an attempt to break the "monotony" of thetext I'm writing, otherwise the text looks like a big wall of words that is unappetizing to read (at least to me :P)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 1 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:508]: Replaced!

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 3 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:521]: Yep, it does compile ;)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 4 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:559]: Because it's not the most straightforward, but it's a good intro to the next bit of text... Let me rephrase this ;)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 5 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:582]: Yes, also fixed to "Flutter web" everywhere else I could find :)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 7 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:590]: Yep, short and sweet.

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 9 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:612]: > Can't you create an iframe with origin trust policies that allow observing cross-frame events?

Technically, if Flutter web could see the `<iframe>`'s [`contentWindow` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/contentWindow), we could `iframe.contentWindow.addEventListener('pointermove'...` (for example) and try to merge events from the iframe with those that are reported to the framework.

In practice, we don't do this in our pointer handling (and I'm not sure if there's a way for users to funnel events from the app to the framework), so the answer to your question is: yes, _with our current implementation_, this is always true.

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 10 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:613]: Replaced by "the contents of the `HtmlElementView`"

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 11 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:628]: I extended this a little bit to introduce what an "overlay" is.

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 14 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:650]: Probably there's a better place, and then we should forward to the better place from here. The thing is that currently this is documented NOWHERE:

https://www.google.com/search?q=%22ensureSemantics%22+site%3Aflutter.dev

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 15 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:652]: They render an `<hr />` that separates the looong description above from the rest of the generated documentation below, which I thought looks better :)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 16 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart]: Nope, just a continuation, it just landed near the ~80 char line and broke it down to a new line.

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 15 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart]: (Removed)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 13 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart]: Removed this paragraph, seems outdated to me.

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 8 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart]: Rephrased.

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 9 above)
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart]: (Also, rephrased the other bits)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 17 above)
[the remark this answers is not in the document]
Comment (ditman) [reply packages/flutter/lib/src/widgets/platform_view.dart:366]: Fixed!

**Review by yjbanov — APPROVED** [review]
lgtm

**Comment by ditman** [comment]
The new documentation is already visible in the `main` branch API docs:

* https://main-api.flutter.dev/flutter/widgets/HtmlElementView-class.html
